src/main/java/ch/epfl/biop/imaris/EasyXT.java

```diff
@@ -774,6 +782,85 @@ public static IDataSetPrx getSurfacesDataset(ISurfacesPrx surface, double downsa
         return data;
     }
 
+    /**
+     * Get an ImagePlus of the spots as a mask (255)
+     *
+     * @param spots , a spots object see {@link  #getSpots(String)}
+     * @return , an ImagePlus
+     * @throws Error
+     */
+    public static ImagePlus getSpotsMask(ISpotsPrx spots) throws Error {
+        return getSpotsImage(spots, false, false);
+    }
+
+    // TODO Comment
+    public static ImagePlus getSpotsLabel(ISpotsPrx spots) throws Error {
+        return getSpotsImage(spots, true, false);
+    }
+
+    // TODO Comment
+    public static ImagePlus getSpotsImage(ISpotsPrx spots, boolean isValueID, boolean isGauss) throws Error {
+        // Prepare the imp to receive spots pixels
+        //
+        // get the calibration info from Imaris
+        ImarisCalibration cal = new ImarisCalibration(app.GetDataSet());
+        // Create a new imp with the cal and get stack ()
+        //ImagePlus imp = IJ.createHyperStack( getOpenImageName(), cal.xSize , cal.ySize, cal.cSize, cal.zSize, cal.tSize , 32);
+        //imp.setCalibration( cal );
+        //ImageStack stack = imp.getStack();
+
+        long[] spots_ids = spots.GetIds();
+        float[][] spots_centerXYZ = spots.GetPositionsXYZ();
+        float[][] spots_radiiXYZ = spots.GetRadiiXYZ();
+        int[] spots_t = spots.GetIndicesT();
+
+        double resXY = cal.pixelWidth;
+        double resZ = cal.pixelDepth;
+        String unit = cal.getUnit();
+
+        // Define default vectors
+        Vector3D V = new Vector3D(1, 0, 0);
+        Vector3D W = new Vector3D(0, 1, 0);
+        if (Math.abs(V.dotProduct(W)) > 0.001) {
+            IJ.log("ERROR : vectors should be perpendicular");
+        }
+
+        int previous_t = spots_t[0];
+        ObjectCreator3D obj = new ObjectCreator3D(cal.xSize, cal.ySize, cal.zSize);
+        obj.setResolution(cal.pixelWidth, cal.pixelDepth, cal.getUnit());
+
+
+        ArrayList<ImagePlus> imps = new ArrayList<ImagePlus>(spots_t[spots_t.length - 1]);
+        // by default the value is 255
+        int val = 255;
+        for (int t = 0; t < spots_t.length; t++) {
+            // if the current spot is from a different time-point
+            if (spots_t[t] != previous_t) {
+                // store the current status into an ImagePlus
+                // N.B. duplicate is required to store the current time-point
+                imps.add(new ImagePlus("t" + previous_t, obj.getStack().duplicate()));
+                // and reset the obj
+                obj.reset();
+            }
+            // but if isValueID is true, use the ID number for the value
+            if (isValueID) val = (int) spots_ids[t];
```

Comment:
General comment: If `isValueID` written in this way, then all variables should be written the same way. Otherwise change `spot_ids` to `spotIds` and all the others as well. We should choose one style and be consistent... But I can take care of that once Romain is done pushing one branch per day :)

Reply:
423fc35f6a779609818c37546fe26bcf61d01a03

Reply:
Thanks! I will go through the rest of the code after this PR and fix the others